### DISCUSSION TITLE
Integrate Dynamic energy prices to calculate charging costs per charge session.

### DISCUSSION THREAD

**Discussion by nderooij** [tillsteinbach/VWsFriend#457] [Ideas]
Like the awesome work on this project man! It's awesome to have proper data about my ID.3 and I can't wait for my car to get 3.0 so I can have the location/trip data filed in as well!

I wanted to raise a feature idea though; What if we could have VWsFriends check the current power price (per hour with a dynamic contract) and use that to calculate the costs of a charging session if using a home charger?

I'm personally in the Netherlands and have the energy tariffs in my home-assistant, so the data is out there. Not a programmer though, so not sure how much work it would be to include this in VWsFriend?

**Comment by tillsteinbach** [comment]
The problem is that the amount charge is estimated based on the SoC. You can later correct it with the real amount. But VWsFriend has no possibility to exactly know what was being charged at a certain time.

**Reply by nderooij** [reply]
So you're saying that data would have to come from the charger and you'd have to implement individual chargers in to VWsFriend?

That makes a lot of sense, thanks for clarifying that man!

Does VWsFriend have an API I can use to have something like Home Assistant push that data to VWs friends to automate this proces?

**Reply by tillsteinbach** [reply]
I have Ideas for a plug-in interface where chargers could be connected with VWsFriend. But right now I have no capacity to do so.

**Reply by nderooij** [reply]
Awesome! Thanks man!
